solutions/33. Search in Rotated Sorted Array.java

```diff
@@ -0,0 +1,38 @@
+//Time complexity: log(n)*log(x)-->this is after calulating mid we have to move over the range in which the element is available.
+//here is we are firstcaluting the mid then checking the which part is sorted then and then we are checking whetehr the element lies in the sorted part or the other //side then we are repeting the same thing till low<=high.
+​
+//Note: I didnt understand one thing that is why we are looking or for the sorted part here. Can you plz explain once.
+​
```

Comment:
it's not that you are searching for the sorted part. Both the sides would be sorted but we are looking for sorted part to make the target search easier by reducing the search space.
input [4,5,6,7,0,1,2], target = 0
search 0 in 4-7 range -> move low pointer to mid + 1 and so on